### ISSUE TITLE
Message tracking is not affected by after_action changes

### ISSUE THREAD

**Issue by chamnap** [ankane/ahoy_email#105]
I have `after_action`, which it overrides the `from` of the message.

```ruby
class ApplicationMailer < ActionMailer::Base
  default from: 'no-reply@domain.com'
end

class Notifications < ApplicationMailer
  after_action  :set_district_sender_email

  def activity_submitted(email, activity)
    @activity = activity
    @district = activity.district

    mail to: email, subject: 'New activity ready to be reviewed!'
  end

  private

  def set_district_sender_email
    return if @district.blank?
    return if @district.sender_email.blank?

    message.from = @district.sender_email
  end
end
```

Basically, I want to override the `from` value (the value is retrieved from `@district` on each action). Everything is working except the tracking. The `from` value is saved with `no-reply@domain.com`. It should be the value received from `@district.sender_email` (`help@domain.com`).

> SQL (7.9ms)  INSERT INTO "ahoy_messages" ("token", "to", "mailer", "subject", "content", "utm_source", "utm_medium", "utm_campaign") VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING "id"  [["token", "2Mttc9jgRNwsSex5D5oQODt4vf9Ikfyo"], ["to", "chamnapchhorn@gmail.com"], ["mailer", "Notifications#activity_submitted"], ["subject", "New activity ready to be reviewed!"], ["content", "Date: Mon, 30 Apr 2018 17:34:31 +0700\r\nFrom: no-reply@domain.com\r\nTo: chamnapchhorn@gmail.com\r\nMessage-ID: <5ae6f13717ade_7e7b3ffc8283f79828794@Chamnaps-MacBook-Pro.local.mail>\r\nSubject: New activity ready to be reviewed!\r\nMime-Version: 1.0\r\nContent-Type: text/html;
....
....
   (5.4ms)  COMMIT
Notifications#activity_submitted: processed outbound mail in 65.5ms
  Ahoy::Message Load (0.6ms)  SELECT  "ahoy_messages".* FROM "ahoy_messages" WHERE "ahoy_messages"."id" = $1 ORDER BY "ahoy_messages"."id" ASC LIMIT $2  [["id", 369915], ["LIMIT", 1]]
   (0.4ms)  BEGIN
  SQL (0.4ms)  UPDATE "ahoy_messages" SET "sent_at" = $1 WHERE "ahoy_messages"."id" = $2  [["sent_at", "2018-04-30 10:34:31.117447"], ["id", 369915]]
   (0.7ms)  COMMIT
Sent mail to chamnapchhorn@gmail.com (39.6ms)
Date: Mon, 30 Apr 2018 17:34:31 +0700
From: help@domain.com
To: chamnapchhorn@gmail.com

**Comment by ankane** [comment]
Hey @chamnap, thanks for reporting 👍 For the next major release, I'm planning to move tracking to an interceptor, which will fix this issue, but I don't think there's an easy fix in the meantime.

**Comment by ankane** [comment]
Hey @chamnap, this is fixed in 1.0.
